### PR TITLE
feat(jira): add support for non-English localization

### PR DESCRIPTION
**Pull request by DmitriyLewen**
## Description
Added support for non-English localization.

## Issues

- aquasecurity/postee/issues/255

### CODE REVIEW

File: README.md
Comment (AndreyLevchenko) [inline README.md:276]: I wouldn't mention particular values. Let's just say "defaults"

**Review by AndreyLevchenko — CHANGES_REQUESTED** [review]

File: README.md
Comment (AndreyLevchenko) [reply README.md:276]: moreover for Jira with not English localization it works exactly same way, but values are localized. So probably it shouldn't be mentioned at all

Comment (simar7) [reply README.md:276]: I thought we would get the field names from the JIRA API and not the user. Would that be better?

Comment (DmitriyLewen) [reply README.md:276]: [priority](https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-priorities/#api-rest-api-2-priority-get) and [issuetype](https://developer.atlassian.com/cloud/jira/platform/rest/v2/api-group-issue-types/#api-rest-api-2-issuetype-get) `id's` do not have a default value for the `High` and `Task` values.  We need to get these values from the user. We get other field names from the JIRA API.

Comment (DmitriyLewen) [reply README.md]: I added the logic for `issueType` and `issuePriority`. You can take a look.

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:67]: Any reason for using this dependency? Standard library should work fine as so:


```
 38  	ts := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
    39  		fmt.Fprintln(w, test.priorities)
    40  	}))
    41  	defer ts.Close()
```

Then use `ts.URL()` as the URL.

Best to not use external dependencies if standard library would meet the need.

Comment (DmitriyLewen) [reply outputs/jira_test.go]: Okay, thanks. I changed the dependency to `httptest`.

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:111]:
```suggestion
			name:          "happy path (fill issueType, jira has 'Bug' field)",
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:117]:
```suggestion
			name:        "bad path (fill issueType, jira doesn't have 'Bug' field)",
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:136]:
```suggestion
				require.NotNil(t, err, test.name)
				assert.Contains(t, err.Error(), test.wantError, test.name)
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:138]:
```suggestion
				require.Equal(t, test.wantIssueType, test.jiraAPI.Issuetype, test.name)
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:190]: This error handling is not necessary.

```suggestion
				w.WriteHeader(200)
				metaInfoJson, _ := json.Marshal(test.metaInfo)
				_, _ = w.Write(metaInfoJson)
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:182]: Will the status code be different if a JIRA error occurs? Does that matter? If it isn't used, just remove the status code altogether to keep tests simple.

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:347]:
```suggestion
				if test.wantError != "" {
					return fmt.Errorf(test.wantError)
				} else {
					return nil
				}
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:360]: nit: 

```suggestion
				require.NotNil(t, err, test.name)
				assert.Contains(t, err.Error(), test.wantError, test.name)
			} else {
				for key, value := range test.wantFieldsConfig {
				        assert.Equal(t, value, fieldsConfig[key], test.name)
				}
```

File: outputs/jira.go
Comment (simar7) [inline outputs/jira.go:369]: Will `ctx.SprintId` be not set if it was non English jira?

File: router/initoutputs_test.go
Comment (simar7) [inline router/initoutputs_test.go:195]: Are these changes related to this PR?

File: outputs/jira.go
Comment (simar7) [inline outputs/jira.go:24]: If it's a non English JIRA, what would happen? Would the first priority be selected from the returned list?

**Review by simar7 — CHANGES_REQUESTED** [review]
left some comments

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go:136]: i don't think that we need these changes
we already have `test.name` in the `Test` field:
```
Error:          Expected value not to be nil.
Test:           TestCreateIssueType/happy_path_(empty_issueType,_jira_have_'Task'_field)
Messages:       happy path (empty issueType, jira have 'Task' field)     // duplication
```

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go:138]: ditto

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: thanks for finding my mistake.

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: thanks for finding my mistake.

File: router/initoutputs_test.go
Comment (DmitriyLewen) [reply router/initoutputs_test.go:195]: i added these tests to improve `router/builders.go` coverage in  codecov bot, after removed [these functions](https://github.com/aquasecurity/postee/pull/271/files/71395a40805263bc8a0b74b51aa2168df797768b#diff-48982bfea0eb5136efb328357261341d8167a96ed1b49c8319afad977998a933L92-L97)

File: outputs/jira.go
Comment (DmitriyLewen) [reply outputs/jira.go:24]: we can't get an analogue of the `high` priority from [rest API](https://github.com/aquasecurity/postee/pull/271/files#diff-76265623be2e5ca53abae86b5d1087ff8bcdc391272004877039fe54435b1f38R312-R319).
If there is no `defaultIssuePriority` in the priority list we will use [the first priority from the getting list](https://github.com/aquasecurity/postee/pull/271/files#diff-76265623be2e5ca53abae86b5d1087ff8bcdc391272004877039fe54435b1f38R312-R323).
User can change the priority in the configuration file, if he needs it.

File: outputs/jira.go
Comment (DmitriyLewen) [reply outputs/jira.go:369]: Sprint is a custom field. The Rest API returns the sprint ID as `customfield{Number}`.
We need more thoughtful logic for sprint.
I think we need to add it a bit later.

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Good idea! thanks!

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Removed

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Done

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Changed! thanks!

File: outputs/jira_test.go
Comment (simar7) [reply outputs/jira_test.go:131]: ok fair enough

File: outputs/jira.go
Comment (simar7) [reply outputs/jira.go:369]: I see - got it

File: outputs/jira.go
Comment (simar7) [reply outputs/jira.go:24]: Roger that

File: router/initoutputs_test.go
Comment (AndreyLevchenko) [inline router/initoutputs_test.go:156]: isn't it a default behavior? Probably `Issuetype` and `Priority` could be just omitted

File: outputs/jira.go
Comment (AndreyLevchenko) [inline outputs/jira.go:276]: 1. verb in function shouldn't be `create` 
2. Check if it's possible to avoid side effects 
3. search loop is implemented twice
4. "don't" -> "doesn't"

File: outputs/jira.go
Comment (AndreyLevchenko) [inline outputs/jira.go:189]: error message needs to be changed as well

File: outputs/jira.go
Comment (AndreyLevchenko) [inline outputs/jira.go:300]: similar notes like to issue type

File: outputs/jira.go
Comment (AndreyLevchenko) [inline outputs/jira.go:344]: also it's not a "create"

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:189]: do the assignment outside `for` loop. Otherwise incorrect value will be set

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:197]: see above

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:144]: Please use `wantResponse` or `wantJiraApiResponse`

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:200]: error conditions are not checked

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:202]: don't use `test.wantJiraApi` just compare `test.boardName` and `jiraApi.BoardName`

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:255]: same as above

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:247]: make -1 a named constant

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go:200]: what error needs to be checked? 
`fetchBoardId` doesn't return error

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:602]: same note as before - assign/restore it once

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:572]: as code is almost same I would create something like `buildHttpHandler(successResponse, errorResponse) func ....`

File: outputs/jira_test.go
Comment (AndreyLevchenko) [inline outputs/jira_test.go:701]: it's not bad just something that was not found :). Also project key can't contain spaces. (just for authenticity)

File: router/initoutputs_test.go
Comment (DmitriyLewen) [reply router/initoutputs_test.go]: Done

File: outputs/jira.go
Comment (DmitriyLewen) [reply outputs/jira.go]: Done

File: outputs/jira.go
Comment (DmitriyLewen) [reply outputs/jira.go]: Changed

File: outputs/jira.go
Comment (DmitriyLewen) [reply outputs/jira.go]: Done

File: outputs/jira.go
Comment (DmitriyLewen) [reply outputs/jira.go]: Done

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Moved

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Moved

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go:206]: We should to check JiraApi, because we also [recording 'boardId' and  'boardType'](https://github.com/aquasecurity/postee/blob/127d5170b3b1624eb3ad6cdc6651cb46b2ee95fb/outputs/jira.go#L70-L76)

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go:242]: Done

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Done

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Done

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Changed

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go]: Changed

**Review by AndreyLevchenko — APPROVED** [review]

**Comment by DmitriyLewen** [comment]
Hello @simar7 
I created some changes. Can you take a look?
Thank you!

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:187]: nit:
```suggestion
	oldCreateClient := createClient
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:573]:
```suggestion
				require.NoError(t, err)
```

File: outputs/jira_test.go
Comment (simar7) [inline outputs/jira_test.go:524]: ditto but for missing project key?

File: outputs/http_test.go
Comment (simar7) [inline outputs/http_test.go:69]: I noticed this as well and fixed it here https://github.com/aquasecurity/postee/pull/287/commits/cc6c7a5d8df1f3aa5e53740180947c540646d406

File: Makefile
Comment (simar7) [inline Makefile:27]: Ideally we should just include all directories like so https://github.com/aquasecurity/postee/pull/287/commits/31bda2aad34e34ac7a74d78d50f540304ec3f399

When I did this, I noticed there are two tests that are making external calls, in this commit I disabled them (for now) until we can find a way to better handle them: https://github.com/aquasecurity/postee/pull/287/commits/31bda2aad34e34ac7a74d78d50f540304ec3f399

File: outputs/jira.go
Comment (DmitriyLewen) [reply outputs/jira.go:378]: You're right! Sprint is a custom field. We get sprint name using the plugin name.

File: outputs/jira_test.go
Comment (DmitriyLewen) [reply outputs/jira_test.go:524]: We test for an empty ProjectKey in [this test](https://github.com/aquasecurity/postee/blob/4e0e0efbeddb93ee56285587da85a74182ff91c6/outputs/jira_test.go#L511-L515).

File: Makefile
Comment (DmitriyLewen) [reply Makefile:27]: If you want to merge this PR after `cc6c7a5` and `31bda2a` merge, I can remove these changes.

**Review by simar7 — APPROVED** [review]

**Pull request merged**
